server/src/sv_main.cpp

```diff
@@ -3287,10 +3292,10 @@ void SV_UpdateSecretCount(player_t& player)
 	    if (&*it == &player)
 	        continue;
 
-	    std::ostringstream buf;
-		
-	    buf << player.userinfo.netname << " found a secret!";
-	    SV_MidPrint(buf.str().c_str(), &(*it), 0);
+		client_t* cl = &(it->client);
+
+		MSG_WriteMarker(&cl->reliablebuf, svc_secretevent);
+		MSG_WriteByte(&cl->reliablebuf, it->id);
```

Comment:
Could you create a function in `msg_server.cpp` to write this out?  It will make it much easier to replace when we upgrade the protocol.